lib/elixir/lib/process.ex

```diff
@@ -548,4 +552,16 @@ defmodule Process do
   @compile {:inline, nillify: 1}
   defp nillify(:undefined), do: nil
   defp nillify(other),      do: other
+
+  defp process_register_error(pid, name) do
+    case {alive?(pid), whereis(name)} do
+      {false, _} ->
+        "Process for #{inspect(pid)} is no longer alive"
+      {true, existing_pid} when is_pid(existing_pid) ->
+        "Process with pid #{inspect(existing_pid)}" <>
```

Comment:
To make this more consistent maybe the error message could be swapped around to be "The name #{inspect(name)} is already registered to process #{inspect(existing_pid)}", what do you think?